generator/rangewriter.go

```diff
@@ -48,15 +57,30 @@ func (rw *RangeWriter) WriteStringLiteral(level int, s string) (r parser.Range,
 		if err != nil {
 			return
 		}
-		if _, err = rw.WriteIndent(level, `_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString("`); err != nil {
-			return
+
+		if rw.extractStrings {
+			index := len(rw.strings)
+			if _, err = rw.WriteIndent(level, "templ_7745c5c3_Err = templ.WriteExtractedString(templ_7745c5c3_Buffer, "+strconv.Itoa(index)+")\n"); err != nil {
+				return
+			}
+		} else {
```

Comment:
When I see if/else to enable/disable a feature, I usually wonder whether I'm missing an abstraction. Should there be a LiteralWriter interface that has two implementations (Production, and WatchMode), or has my mind been warped by too much C#. 😁 

I'm happy with whatever you think best here.

Reply:
Took your suggestion and used an interface with implementations. Works nicely 👌🏾